sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FieldValueType.java

```diff
@@ -3,62 +3,145 @@
-    FLOAT("float"),
+    /**
+     * Static value time for FieldValueType.
+     */
+    public static final FieldValueType TIME = fromString("time");
 
-    /** Enum value integer. */
-    INTEGER("integer"),
+    /**
+     * Static value phone number for FieldValueType.
+     */
+    public static final FieldValueType PHONE_NUMBER = fromString("phoneNumber");
 
-    /** Enum value array. */
-    LIST("list"),
+    /**
+     * Static value float for FieldValueType.
+     */
+    public static final FieldValueType FLOAT = fromString("number");
 
-    /** Enum value object. */
-    MAP("map");
+    /**
+     * Static value integer for FieldValueType.
+     */
+    public static final FieldValueType INTEGER = fromString("integer");
 
-    /** The actual serialized value for a FieldValueType instance. */
-    private final String value;
+    /**
+     * Static value list for FieldValueType.
+     */
+    public static final FieldValueType LIST = fromString("array");
 
-    FieldValueType(String value) {
-        this.value = value;
-    }
+    /**
+     * Static value map for FieldValueType.
+     */
+    public static final FieldValueType MAP = fromString("object");
 
     /**
-     * Parses a serialized value to a FieldValueType instance.
+     * Parses a serialized value to a {@link FieldValueType} instance.
      *
      * @param value the serialized value to parse.
+     *
      * @return the parsed FieldValueType object, or null if unable to parse.
      */
-    @JsonCreator
     public static FieldValueType fromString(String value) {
-        FieldValueType[] items = FieldValueType.values();
-        for (FieldValueType item : items) {
-            if (item.toString().equalsIgnoreCase(value)) {
-                return item;
-            }
-        }
-        return null;
+        return fromString(value, FieldValueType.class);
     }
 
-    @JsonValue
-    @Override
-    public String toString() {
-        return this.value;
+    /**
+     * Converts the form field value to a specific expandable string enum type.
+     *
+     * @param formField The recognized field value that needs to be converted.
+     * @param <T> the class of the field.
+     *
+     * @return the converted value of the recognized field.
+     * @throws UnsupportedOperationException if the {@code formField} type does not match the casting value type.
+     * @throws NullPointerException if {@code formField} is {@code null}
+     */
+    @SuppressWarnings({"unchecked"})
+    public <T> T cast(FormField<?> formField) {
+        Objects.requireNonNull(formField, "'formField' cannot be null");
+        if (DATE.equals(this)) {
+            if (DATE == formField.getValueType()) {
+                return (T) formField.getValue();
+            } else {
+                logger.logExceptionAsError(new UnsupportedOperationException(String.format("Unsupported cast operation "
+                    + "or field value of type: %s", formField.getValueType())));
+            }
```

Comment:
Update the error message to include both field type and the attempted cast type.
`String.format("Cannot cast from field value of type %s to type %s", formField.getValueType(), DATE)`